### PR TITLE
Improving cancellation flow

### LINKED ISSUE #5082: [CAL-184] Cancel a booking flow needs updating
Current flow:

[9962697e-7f84-47e9-99d6-39df030209ab](https://uploads.linear.app/e86bf957-d82f-465e-b205-135559f4b623/4bedcfa7-c406-4873-a9cf-2044d0255c7c/9962697e-7f84-47e9-99d6-39df030209ab)

Flow should be: 

[containers - ❖ Cal DS v2.0 (Figma)](https://www.figma.com/file/9MOufQNLtdkpnDucmNX10R/%E2%9D%96-Cal-DS-v2.0?node-id=17954%3A199476)

> From [Linear-GitHub Sync](https://synclinear.com)

### LINKED ISSUE #5431: "Need to make a change? - Cancel" on recurring events, deletes just the next upcoming event
Found a bug? Please fill out the sections below. 👍

### Issue Summary

"Need to make change? - Cancel" on recurring events, deletes just the next upcoming event instead of the entire series. 

### Steps to Reproduce

1. Schedule a recurring event
2. After creating the event click on "Cancel" next to "Need to make a change".
3. It cancels the next upcoming event instead of the entire series of events.
4. The similar behavior when clicking on an existing recurring event in Bookings -> Recurring, and opens a similar page with the same option to Cancel next to "Need to make a change?"

Screenshot from 2022-11-09 01-06-11

### Technical details

Self hosted, main branch, v2.1.3

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

- Removes old /cancel/[uid] page and uses /success page for all cancellations (with cancellation part already opened up and scrolled down)
- Old links (cancel/[uid]) that are still in (old) emails are redirected to new /success page with open cancellation part
- Fixes cancellations of recurring events 
  - You can either cancel all remaining events or just one specific event
  - Cancel link from email, redirects to /succes page to cancel all remaining events
  
Fixes #5431
Fixes #5082

Loom Video: https://www.loom.com/share/104cf5c9f47f481b9784109c8cb5e21f

**Environment**: Staging(main branch)

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by zomars** [comment]
Before reviewing, given that the success page is being used for more purposes it may start to make sense to have a dedicated booking url like `/booking/[uid]` And there you can see that booking info and reschedule and cancel as well.

File: apps/web/pages/success.tsx [hunk 1]
```diff
@@ -35,9 +35,8 @@ import { localStorage } from "@calcom/lib/webstorage";
 import prisma, { baseUserSelect } from "@calcom/prisma";
 import { Prisma } from "@calcom/prisma/client";
 import { EventTypeMetaDataSchema } from "@calcom/prisma/zod-utils";
-import Button from "@calcom/ui/Button";
 import { Icon } from "@calcom/ui/Icon";
-import { EmailInput } from "@calcom/ui/form/fields";
+import { Button, EmailInput } from "@calcom/ui/components";
 
 import { asStringOrThrow } from "@lib/asStringOrNull";
```
Comment (zomars) [inline apps/web/pages/success.tsx:41]:
```suggestion
```

File: apps/web/pages/success.tsx [hunk 2]
```diff
@@ -171,7 +192,7 @@ export default function Success(props: SuccessProps) {
   const isEmbed = useIsEmbed();
   const shouldAlignCentrallyInEmbed = useEmbedNonStylesConfig("align") !== "left";
   const shouldAlignCentrally = !isEmbed || shouldAlignCentrallyInEmbed;
-  const [isCancellationMode, setIsCancellationMode] = useState(false);
+  const [isCancellationMode, setIsCancellationMode] = useState(cancel);
```
Comment (zomars) [inline apps/web/pages/success.tsx:195]: Can we just use the state from the URL directly?

**Review by zomars — COMMENTED** [review]
Code looking good. Testing locally...

**Review by zomars — APPROVED** [review]
All good here! Made some improvements and it's good to go.


LGTM

**Comment by Jaibles** [comment]
> Before reviewing, given that the success page is being used for more purposes it may start to make sense to have a dedicated booking url like /booking/[uid] And there you can see that booking info and reschedule and cancel as well.

👆This sounds ideal. Every booking should have a link that shows the current state of the booking whether its scheduled, cancelled or in the past etc.

**Comment by JeroenReumkens** [comment]
We pretty much already have this url 🔥  https://app.cal.com/success?uid=dA3ipZ36dYmaGEuNyBjb8z 
Url could be better though, and I see that it does NOT adapt to whether the meeting is for example in the past. 

But I guess we could use this as a nice starting point. And the /success path could easily be renamed to /booking/... since even as a success step after booking I think it makes sense to go to the /booking page. 

This UUID part has been added recently by @mischarouleaux.

**Pull request merged**